### ISSUE TITLE
🐛 [BUG] - Datadog backend - bad_event_counts spikes

### ISSUE THREAD

**Issue by maximepln** [google/slo-generator#343]
### SLO Generator Version

image: slo-generator:2.3.4 

### Python Version

3+

### What happened?

When using datadog as a backend with the slo-generator, we found out that multiple time a day, the value of bad_event_counts was incoherent with what we see on datadog triggering our duty with false positive incident. 

As a example, the following SLI had a spike of bad_event_count (26) today at 11:51am

<img width="1215" alt="console-screenshot" src="https://github.com/google/slo-generator/assets/56535266/41fdc4eb-e8df-43e6-9a06-26a501264fa0">

When checking the datadog metrics we don't see a single bad event: 

<img width="1652" alt="Screenshot 2023-06-12 at 15 45 05" src="https://github.com/google/slo-generator/assets/56535266/30f88294-4e0b-4714-b941-3b63f7b78f88">

Here is my slo-generator datadog set-up 

```
  method: good_bad_ratio
  service_level_indicator:
    query_good: SecondLife_back_all_calls{http.method:get,http.status_code:200,http.url_details.path:/v1/price_estimations/suggest}
    query_valid: SecondLife_back_all_calls{http.method:get,!http.status_code:4*,http.url_details.path:/v1/price_estimations/suggest}
```

Note: Slo-generator is configured to request metrics from datadog every minutes, most of the time this issue occurs for a single minute before going back to normal.

### What did you expect?

We are expecting that the slo-generator output is equivalent to what we are getting in Datadog. 

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by maximepln** [comment]
A possible workaround if finding the source is too complex, would be to authorise `query_bad` instead of `query_valid` with Datadog backend, it will helps avoid having spikes of bad_event_counts.

**Comment by bkamin29** [comment]
We suspect that the root-cause is related to the data consistency on Datadog side.
We get good events and then we get valid events.
Theses 2 request are probably not taken in charge by the same endpoint on Datadog side, so if theses 2 endpoint don't have exactly the same data freshness, it's cause this issue.
First call you get 9 good event, and second call you get 10 valid event => 9-10, result is 1 bad event on SLO Generator side but probably 0 on Datadog

**Comment by lvaylet** [comment]
closed by #377
